src/LinuxTracing/SwitchesStatesNamesVisitor.cpp

```diff
@@ -189,6 +204,17 @@ void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
   }
 }
 
+void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
+                                       const SchedWakeupPerfEventData& event_data) {
+  VisitSchedWakeup(event_timestamp, event_data, /*wait_for_callstack=*/false);
+}
+
+void SwitchesStatesNamesVisitor::Visit(uint64_t event_timestamp,
+                                       const SchedWakeupWithStackPerfEventData& event_data) {
+  const bool wait_for_callstack = event_data.data != nullptr;
```

Comment:
Same here, I don't understand why we would use `SchedWakeupWithStackPerfEventData` and set `event_data.data == nullptr` instead of using `SchedWakeupPerfEventData`.

Reply:
As above.